### PR TITLE
Issue236

### PR DESCRIPTION
**Pull request by dharanisakary**

### CODE REVIEW

**Comment by jonwetzel** [comment]
@dharanisakary merge conflict detected

File: www/js/controller.js [hunk 1]
```diff
@@ -594,16 +594,27 @@ define([
...
+					var variable_name = this._model.authored.getName(desc.value);
+					var option = {label:variable_name, value: variable_name};
+					if(v.getOptions(option)){
+						var disabled = false;
+						if(exists!== null && model.getName(exists)!== undefined)
+							disabled = true
+						v.getOptions(option).disabled=disabled;
+					}
+					// To reflect the changes to UI
+					v.startup();
```
Comment (sachingrover211) [inline www/js/controller.js]: add the logic to disable the list in model.js checking which are nodes exist. If not then we should look at why it wasnt working earlier? I dont understand it seems that you are doing the whole logic again. Same logic that has been done just above it.

File: www/js/controller.js [hunk 2]
```diff
@@ -596,24 +596,19 @@ define([
 				var d = registry.byId(this.controlMap.description);
 				var v = registry.byId(this.controlMap.variable);
 				array.forEach(this._model.authored.getDescriptions(), lang.hitch(this, function(desc){
-					var exists =  model.getNodeIDFor(desc.value);
-
-					if( d.getOptions !== undefined && d.getOptions(desc)) {
-						d.getOptions(desc).disabled=exists;
-						if(desc.value == nodeName){
-							d.getOptions(desc).disabled=false;
+					if(this._model.doesStudentNodeExist(desc.value)){
```
Comment (sachingrover211) [inline www/js/controller.js]: this code will be called for author mode as well. what happens then?

Comment (sachingrover211) [reply www/js/controller.js]: I am saying as this is in controller.js

Comment (dharanisakary) [reply www/js/controller.js]: This part is never executed in author mode as model.getNodeIDFor is not defined for author

**Comment by sachingrover211** [comment]
@jonwetzel this PR is really old. Isnt this already happening?

**Comment by jonwetzel** [comment]
The issue was resolved by other code.

**Pull request closed without merging**